sdk/iothub/Microsoft.Azure.Management.IotHub/tests/ScenarioTests/IotHubLifeCycleTests.cs

```diff
@@ -2,430 +2,519 @@
 // Licensed under the MIT License. See License.txt in the project root for
 // license information.
 
+using FluentAssertions;
+using IotHub.Tests.Helpers;
+using Microsoft.Azure.Management.IotHub;
+using Microsoft.Azure.Management.IotHub.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Rest.Azure;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
 namespace IotHub.Tests.ScenarioTests
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Threading;
-    using IotHub.Tests.Helpers;
-    using Microsoft.Azure.Management.IotHub;
-    using Microsoft.Azure.Management.IotHub.Models;
-    using Microsoft.Azure.Management.ResourceManager;
-    using Microsoft.Azure.Management.ResourceManager.Models;
-    using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
-    using Xunit;
-
     public class IotHubLifeCycleTests : IotHubTestBase
     {
         [Fact]
-        public void TestIotHubCreateLifeCycle()
+        public async Task TestIotHubCreateLifecycle()
         {
-            using (MockContext context = MockContext.Start(this.GetType()))
+            try
             {
-                this.Initialize(context);
+                using var context = MockContext.Start(GetType());
+                Initialize(context);
 
                 // Create Resource Group
-                var resourceGroup = this.CreateResourceGroup(IotHubTestUtilities.DefaultResourceGroupName);
+                ResourceGroup resourceGroup = await CreateResourceGroupAsync(IotHubTestUtilities.DefaultResourceGroupName)
+                    .ConfigureAwait(false);
 
                 // Check if Hub Exists and Delete
-                var operationInputs = new OperationInputs()
-                {
-                    Name = IotHubTestUtilities.DefaultIotHubName
-                };
-
-                var iotHubNameAvailabilityInfo = this.iotHubClient.IotHubResource.CheckNameAvailability(operationInputs);
+                IotHubNameAvailabilityInfo iotHubNameAvailabilityInfo = await iotHubClient.IotHubResource
+                    .CheckNameAvailabilityAsync(IotHubTestUtilities.DefaultIotHubName)
+                    .ConfigureAwait(false);
 
                 if (!(bool)iotHubNameAvailabilityInfo.NameAvailable)
```

Comment:
what is the data type of NameAvailable? string? that's just odd

Reply:
bool?

Reply:
it looks like it's true, on line 51 we are checking if it's true and I think it has to be boolean for `BeTrue()` to be available.

Reply:
the reason I ask is that there is some casting happening that I am not sure is necessary

Reply:
It is a nullable bool. It is being cast to a normal bool.

The alternative is to do `if (iotHubNameAvailabilityInfo.NameAvailable.HasValue && !iotHubNameAvailabilityInfo.NameAvailable.Value)`